snapcraft/main.py

```diff
@@ -110,6 +113,8 @@ def main():
 
     log.configure(log_level=log_level)
 
+    common.set_enable_parallel_builds(not args['--no-parallel-build'])
+
     try:
         commands.load(cmd).main(argv=args['ARGS'])
```

Comment:
It would need a rename, but this should live in the `yaml` module which loads the `project` (there, that's a new name) and keeps the state of it (state which does not correspond to an individual plugin).